src/core/cpu_mesh.rs

```diff
@@ -372,12 +430,21 @@ impl CPUMesh {
                 indices.push(((i + 1) * angle_subdivisions + j) as u16);
             }
         }
-        let mut mesh = Self {
-            name: "cone".to_string(),
-            positions,
-            indices: Some(Indices::U16(indices)),
-            ..Default::default()
-        };
+
+        gen.generate(Some(Indices::U16(indices)), name, &vertices)
+    }
+
+    ///
+    /// Returns a cone mesh around the x-axis in the range `[0..1]` and with radius 1 at -1.0.
+    ///
+    pub fn cone(angle_subdivisions: u32) -> Self {
+        let mut mesh = Self::cylinder_gen("cone", angle_subdivisions, CPUMeshGenerator {
+            positions: &mut |v| [v.x, v.y * (1.0 - v.x), v.z * (1.0 - v.x)],
+            normals: None,
+            tangents: None,
+            uvs: None,
+            colors: None,
+        });
```

Comment:
I was a bit lazy here, didn't verify if the cylinder and cone functions are really the same except with the `(1.0 - v.x)` multiplier.